libnavigation-ui/src/main/java/com/mapbox/navigation/ui/routealert/RouteAlertToll.kt

```diff
@@ -0,0 +1,120 @@
+package com.mapbox.navigation.ui.routealert
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import androidx.core.content.ContextCompat
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.style.expressions.Expression
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory
+import com.mapbox.mapboxsdk.style.layers.PropertyValue
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import com.mapbox.navigation.base.trip.model.alert.RouteAlert
+import com.mapbox.navigation.base.trip.model.alert.TollCollectionAlert
+import com.mapbox.navigation.base.trip.model.alert.TollCollectionType
+import com.mapbox.navigation.ui.R
+import com.mapbox.navigation.ui.routealert.MapboxRouteAlert.Companion.getMapboxRouteAlertSymbolLayerProperties
+
+/**
+ * Handle [TollCollectionAlert] to add an icon on the map to indicate where a toll is presented.
+ *
+ * @param context to retrieve drawable/string resources
+ * @param style to add source/layer into
+ * @param drawable the icon which represents a toll alert. Use default icon if not assigned
+ * @param properties customized properties. Don't customize the 'iconImage' and 'textField'
+ *  properties. These two properties use the default settings.
+ */
+
+class RouteAlertToll @JvmOverloads constructor(
```

Comment:
Decompile to bytecode: 
```
   @JvmOverloads
   public RouteAlertToll(@NotNull Context context, @NotNull Style style, @NotNull Drawable drawable, @NotNull PropertyValue... properties) {
    ...
   }

   // $FF: synthetic method
   public RouteAlertToll(Context var1, Style var2, Drawable var3, PropertyValue[] var4, int var5, DefaultConstructorMarker var6) {
      if ((var5 & 4) != 0) {
         Drawable var10000 = ContextCompat.getDrawable(var1, drawable.mapbox_ic_route_alert_toll);
         Intrinsics.checkNotNull(var10000);
         var3 = var10000;
      }

      if ((var5 & 8) != 0) {
         var4 = MapboxRouteAlert.Companion.getMapboxRouteAlertSymbolLayerProperties();
      }

      this(var1, var2, var3, var4);
   }

   @JvmOverloads
   public RouteAlertToll(@NotNull Context context, @NotNull Style style, @NotNull Drawable drawable) {
      this(context, style, drawable, (PropertyValue[])null, 8, (DefaultConstructorMarker)null);
   }

   @JvmOverloads
   public RouteAlertToll(@NotNull Context context, @NotNull Style style) {
      this(context, style, (Drawable)null, (PropertyValue[])null, 12, (DefaultConstructorMarker)null);
   }
```

seems that java and kotlin users have a bit different API with `@JvmOverloads`: java's users cannot use constructor like `constructor(context, style, properties)`. 

I would suggest use `Builder` here to avoid inconsistency

Reply:
> seems that java and kotlin users have a bit different API with `@JvmOverloads`: java's users cannot use constructor like `constructor(context, style, properties)`. 

👍  that's true. Let me fix this.